src/main/java/competition/subsystems/lights/LightsCommunicationSubsystem.java

```diff
@@ -109,7 +112,9 @@ public void periodic() {
         // Figure out what we want to send to the arduino
         if (!dsEnabled) {
             currentState = LightsStateMessage.RobotDisabled;
-        } else if (collector.getGamePieceCollected()) {
+        }else if(arm.isMaintainerAtGoal()){
+            currentState = LightsStateMessage.ArmAtTargetPosition;
+        }else if (collector.getGamePieceCollected()) {
```

Comment:
There is essentially a priority order here - things "higher" on the else-if chain will supersede ones below. Given that the arm is "at target position" nearly all the time, except for brief periods when it is commanded with new goals, this will almost always be showing - and crowding out information like getGamePieceCollected.

As a result, I suggest putting this near the bottom of the list.